File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/templates/TemplateOverlayTop.kt
package com.perrystreet.woof.designsystem.atomic.templates

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.WindowInsetsSides
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.only
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.safeDrawing
import androidx.compose.foundation.layout.windowInsetsPadding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.PaddingRoles

@Composable
fun TemplateOverlayTop(content: @Composable () -> Unit) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .windowInsetsPadding(WindowInsets.safeDrawing.only(WindowInsetsSides.Top + WindowInsetsSides.Horizontal))
            .padding(
                top = PaddingRoles.Screen.Expanded.dp,
                start = PaddingRoles.Screen.Regular.dp,
                end = PaddingRoles.Screen.Regular.dp,
            ),
        contentAlignment = Alignment.TopCenter,
    ) {
        content()
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Templates/TemplateOverlayTop.swift
import SwiftUI

public struct TemplateOverlayTop<Content: View>: View {
    private let content: () -> Content

    public init(@ViewBuilder content: @escaping () -> Content) {
        self.content = content
    }

    public var body: some View {
        ZStack(alignment: .top) {
            content()
        }
        .padding(.top, PaddingRoles.Screen.expanded.rawValue)
        .padding(.horizontal, PaddingRoles.Screen.regular.rawValue)
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .top)
        .allowsHitTesting(false)
    }
}
